### PR TITLE
ship ax_check_compile_flag.m4 with source code

### PR DESCRIPTION
**Pull request by drossetti**
ax_check_compile_flag.m4 contains the implementation of AX_CHECK_COMPILE_FLAG.
This simplifies installation a little bit

### CODE REVIEW

File: m4/ax_check_compile_flag.m4 [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#   NOTE: Implementation based on AX_CFLAGS_GCC_OPTION. Please keep this
+#   macro in sync with AX_CHECK_{PREPROC,LINK}_FLAG.
+#
+# LICENSE
+#
+#   Copyright (c) 2008 Guido U. Draheim <guidod@gmx.de>
+#   Copyright (c) 2011 Maarten Bosmans <mkbosmans@gmail.com>
+#
+#   Copying and distribution of this file, with or without modification, are
+#   permitted in any medium without royalty provided the copyright notice
```
Comment (pakmarkthub) [inline m4/ax_check_compile_flag.m4]: Do we need the copyright notice?

**Comment by drossetti** [comment]
@pakmarkthub I merged the fix by mistake, I apologize

**Comment by drossetti** [comment]
I've just reverted the merge

File: m4/ax_check_compile_flag.m4 (hunk 1 above)
Comment (drossetti) [reply m4/ax_check_compile_flag.m4]: I guess we do, that was the author wish

**Comment by drossetti** [comment]
@spotluri the commit shows as merged so creating a new PR does not show up as an option.
I can rewrite the history of master branch to get rid of the last 1 or 2 commits, if that makes sense, or simply revert the revert (last commit) if you signoff the review.

**Comment by spotluri** [comment]
@drossetti, the PR looks good to me. I have verified that AX_CHECK_COMPILE_FLAG error goes away. 

Reverting the last commit sounds good.

**Pull request merged**